src/vcSettingsUI.cpp

```diff
@@ -517,7 +517,29 @@ void vcSettingsUI_Show(vcState *pProgramState)
 
             if (pProgramState->settings.convertdefaults.watermark.pTexture != nullptr)
             {
-              ImGui::Image(pProgramState->settings.convertdefaults.watermark.pTexture, ImVec2(256, 256));
+              //Since we're allowing images of any dimensions, we need to make sure it fits in the UI.
+              int dimension[2] = { pProgramState->settings.convertdefaults.watermark.width,
+                                   pProgramState->settings.convertdefaults.watermark.height };
+
+              int maxDimension[2] = {512, 256};
+              int minDimension = 2;
+
+              for (int i = 0; i < 2; i++)
+              {
+                if (dimension[i] > maxDimension[i])
+                {
+                  float factor = float(dimension[i]) / maxDimension[i];
+                  dimension[i] = int(float(dimension[i]) / factor);
+                  int j = (i + 1) % 2;
+                  dimension[j] = int(float(dimension[j]) / factor);
+
+                  //Is the other dimension too thin now?
+                  if (dimension[j] < minDimension)
+                    dimension[j] = minDimension;
```

Comment:
Doesn't this break the aspect ratio? Though I guess anyone with a `512+:1` or `1:256+` ratio "watermark" isn't really going to have a fun time at all.

Reply:
Yes it does. My intent is to display something, even if the aspect ratio of the image is such that one dimension vanishes after resizing to fit in the UI. Perhaps another solution could be to simply enlarge and clip such images, thus only showing a portion of the watermark, but still retaining the correct aspect ratio?

Reply:
Also note this is only for the preview in settings, not the actual watermark

Reply:
I'm not too fussed on this - I'm not sure it's a realistic situation anyway. This comment probably didn't need to be made, I just wanted confirmation that the aspect ratio was intentionally broken.